# HG changeset patch
# User Kai Habel <kai.habel@gmx.de>
# Date 1296229982 -3600
#      Fri Jan 28 16:53:02 2011 +0100
# Node ID 03c7fdee3d36e01d362bfe32da8a807fb96161c9
# Parent  49f1b77a4911467873c8dfcdf55d90d912b576b7
Make file filter value 1-based now

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-28  Kai Habel  <kai.habel@gmx.de>
+
+	* DLD-FUNCTIONS/__fltk_uigetfile__.cc: File filter value is 1-based
+now.
+
 2011-01-28  John W. Eaton  <jwe@octave.org>
 
 	* pr-output.cc (Foutput_max_field_width): Set internal variable
 	output_max_field_width, not output_precision.
 
 2011-01-28  John W. Eaton  <jwe@octave.org>
 
 	* pr-output.cc (Fformat, Ffixed_point_format, Foutput_precision,
diff --git a/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc b/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc
--- a/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc
+++ b/src/DLD-FUNCTIONS/__fltk_uigetfile__.cc
@@ -108,17 +108,17 @@ Undocumented internal function.\n\
           retval(0) = file_cell;
         }
 
       if (multi_type == Fl_File_Chooser::DIRECTORY)
         retval(0) = std::string (fc->value ());
       else
         {
           retval(1) = std::string (fc->directory ()) + sep;
-          retval(2) = fc->filter_value ();
+          retval(2) = fc->filter_value () + 1;
         }
     }
 
   fc->hide ();
   Fl::flush ();
   delete fc;
 
   return retval;
